File: backend/frontend/src/types/prediction.ts
export type PredictionResponse = {
  success: boolean
  filename: string
  prediction_label: "REAL" | "FAKE" | string
  raw_prediction: number
  fake_probability: number
  real_probability: number
  confidence: number
  threshold_used: number
  explanation: string
  heatmap_path: string | null
  // ONNX model extended fields (present when new model is active)
  fake_type: string | null
  fake_type_confidence: number | null
  type_probabilities: Record<string, number> | null
}

export type PredictionRecord = {
  id: number
  user_id: number | null
  original_filename: string
  prediction_label: string
  raw_prediction: number
  fake_probability: number
  real_probability: number
  confidence: number
  explanation: string | null
  heatmap_path: string | null
  created_at: string
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionResponse(BaseModel):
    success: bool
    filename: str
    prediction_label: str
    raw_prediction: float = Field(ge=0.0, le=1.0)
    fake_probability: float = Field(ge=0.0, le=100.0)
    real_probability: float = Field(ge=0.0, le=100.0)
    confidence: float = Field(ge=0.0, le=100.0)
    threshold_used: float
    explanation: str
    heatmap_path: str | None = None
    # ONNX model extended fields
    fake_type: str | None = None
    fake_type_confidence: float | None = None
    type_probabilities: dict | None = None
    # Multi-face detection fields
    faces_detected: int = 1
    flagged_face_index: int | None = None

